pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1681,3 +1691,56 @@ func isDeviceOpened(deviceToDetach AttachedVolume, hostUtil hostutil.HostUtils)
 	}
 	return deviceOpened, nil
 }
+
+func cancelControlPlaneExpansion(pvc *v1.PersistentVolumeClaim, pv *v1.PersistentVolume, client clientset.Interface) (bool, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) {
+		return false, nil
+	}
+
+	newSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+	statusSize := pvc.Status.Capacity[v1.ResourceStorage]
+	pvSize := pv.Spec.Capacity[v1.ResourceStorage]
+	if statusSize.Cmp(newSize) >= 0 && pvSize.Cmp(newSize) >= 0 {
+		klog.Infof("PVC %s already is of requested size", util.GetPersistentVolumeClaimQualifiedName(pvc))
```

Comment:
If newSize > statusSize, then this message is not accurate.

Reply:
if `newSize > statusSize` then there is no reason to mark volume expansion as finished. In that case volume expansion will still be tried.